packages/p2p/src/stores/general-store.js

```diff
@@ -335,14 +337,59 @@ export default class GeneralStore extends BaseStore {
 
         this.updateP2pNotifications(notifications);
     }
-
     redirectToOrderDetails(order_id) {
         const { order_store } = this.root_store;
         this.redirectTo('orders');
         this.setOrderTableType(order_list.INACTIVE);
         order_store.setOrderId(order_id);
     }
 
+    async showAdTypeChangedNotification() {
+        const { floating_rate_store, my_ads_store } = this.root_store;
+
+        await my_ads_store.loadMoreAds({ startIndex: 0 });
+        if (floating_rate_store.change_ad_alert) {
+            if (floating_rate_store.rate_type === ad_type.FLOAT) {
+                if (floating_rate_store.reached_target_date) {
+                    this.props.addNotificationMessage({
+                        header: <Localize i18n_default_text='Your fixed rate ads are deactivated' />,
+                        message: (
+                            <Localize i18n_default_text='You can still find them in My ads. Switch to floating rates to reactivate.' />
+                        ),
+                        key: 'floating-to-fixed-ad-deactivated',
+                        platform: 'P2P',
+                        type: 'p2p_ad_type_changed',
+                    });
+                } else {
+                    this.props.addNotificationMessage({
+                        header: <Localize i18n_default_text='Floating rates are enabled' />,
+                        message: (
+                            <Localize
+                                i18n_default_text='Please switch your ads to floating rates by {{end_date}}.'
+                                values={{
+                                    end_date: floating_rate_store.fixed_rate_adverts_end_date || '',
```

Comment:
do we expect `end_date` to be undefined here? the message doesn't look right if `end_date` is empty